astropy/coordinates/builtin_frames/hadec.py

```diff
@@ -0,0 +1,110 @@
+        The proper motion in declination for this object (``pm_ha_cosdec`` must
+        also be given).
+    radial_velocity : `~astropy.units.Quantity`, optional, keyword-only
+        The radial velocity of this object."""
+
+doc_footer = """
+    Other parameters
+    ----------------
+    obstime : `~astropy.time.Time`
+        The time at which the observation is taken.  Used for determining the
+        position and orientation of the Earth.
+    location : `~astropy.coordinates.EarthLocation`
+        The location on the Earth.  This can be specified either as an
+        `~astropy.coordinates.EarthLocation` object or as anything that can be
+        transformed to an `~astropy.coordinates.ITRS` frame.
+    pressure : `~astropy.units.Quantity`
+        The atmospheric pressure as an `~astropy.units.Quantity` with pressure
+        units.  This is necessary for performing refraction corrections.
+        Setting this to 0 (the default) will disable refraction calculations
+        when transforming to/from this frame.
+    temperature : `~astropy.units.Quantity`
+        The ground-level temperature as an `~astropy.units.Quantity` in
+        deg C.  This is necessary for performing refraction corrections.
+    relative_humidity : `~astropy.units.Quantity` or number.
+        The relative humidity as a dimensionless quantity between 0 to 1.
+        This is necessary for performing refraction corrections.
+    obswl : `~astropy.units.Quantity`
+        The average wavelength of observations as an `~astropy.units.Quantity`
+         with length units.  This is necessary for performing refraction
+         corrections.
+
+    Notes
+    -----
+    The refraction model is based on that implemented in ERFA, which is fast
+    but becomes inaccurate for altitudes below about 5 degrees.  Near and below
+    altitudes of 0, it can even give meaningless answers, and in this case
+    transforming to HADec and back to another frame can give highly discrepant
+    results.  For much better numerical stability, leave the ``pressure`` at
+    ``0`` (the default), thereby disabling the refraction correction and
+    yielding "topocentric" equatorial coordinates.
+    """
+
+
+@format_doc(base_doc, components=doc_components, footer=doc_footer)
+class HADec(BaseCoordinateFrame):
+    """
+    A coordinate or frame in the Hour Angle-Declination system (Equatorial
+    coordinates) with respect to the WGS84 ellipsoid.  Hour Angle is oriented
+    with respect to upper culmination such that the hour angle is negative to
+    the East and positive to the West.
+
+    This frame is assumed to *include* refraction effects if the ``pressure``
+    frame attribute is non-zero.
+
+    The frame attributes are listed under **Other Parameters**, which are
+    necessary for transforming from HADec to some other system.
+    """
+
+    frame_specific_representation_info = {
+        r.SphericalRepresentation: [
+            RepresentationMapping('lon', 'ha'),
```

Comment:
In the example I posted, I noticed that the units for `HA` are degrees, which is perhaps not what we want... One can change it by here using `RepresentationMapping('lon', 'ha', u.hourangle)`

Reply:
Yes, I noticed that also. Hourangle is definitely more appropriate here.

Reply:
Can I just pull from my branch to get your changes? Or does the rebase interfere with that somehow? I've never done a rebase myself.

Reply:
Thanks. That appears to have worked.